contributors/design-proposals/auth/kubectl-exec-plugins.md

```diff
@@ -0,0 +1,193 @@
+# Out-of-tree client authentication providers
+
+Author: @ericchiang
+
+# Objective
+
+This document describes a credential rotation strategy for kubectl and client-go
```

Comment:
Just wanted to call out that we would like to be able to use this for kube-proxy and kubelet as well, to authenticate nodes with AWS IAM credentials.

Reply:
Can you expand a little on that? Why would you want to give kube-proxy IAM credentials to talk to the API server instead of a regular service account?

Reply:
if it's supported by client-go, then kubeproxy and kubelet would pick it up automatically (though kubelet would not be able to take advantage of the nodeauthorizer/noderestriction components unless an IAM-to-node-name mapping could be determined)

Reply:
The stress on kubectl in this proposal is a bit misleading. The change would necessarily be in client-go, thus supported by any component that installs the plugin. Why call out kubectl?

Reply:
> Why call out kubectl?

Most of the use cases I was thinking about were about kubectl users. But you're right this should be more general to client-go.

Reply:
@ericchiang, our node identity is essentially IAM credentials that are present on the node, so it makes sense to us to use those directly for authentication.  If its already in client-go, why not use it?

Reply:
@liggitt Right, the heptio authenticator provides the ability to map IAM role to node name.

Reply:
> @liggitt Right, the heptio authenticator provides the ability to map IAM role to node name.

ah, great